tests/integration/test_automation/testParser.py

```diff
@@ -0,0 +1,507 @@
+                del self.theList[l]
+                message_found = True
+                break
+        if certainty == False:
+            return True
+        if message_found == True:
+            return True
+        else:
+            return False
+
+    def isListEmpty(self):
+        isEmpty = False
+        if len(self.theList) == 0:
+            isEmpty = True
+        else:
+            isEmpty = False
+
+        return isEmpty         
+
+
+class AXI_LIST_CHECKER:
+
+    def __init__(self, theList, rw):
+        self.theList = theList
+        self.rw = rw
+ 
+
+    def checker (self, group_index, message_index, certainty):
+        message_found = False
+        for l in range(0,len(self.theList)):         
+            if self.theList[l].address == search_address :
+                log_file.write(self.theList[l].srcline)
+                del self.theList[l]
+                message_found = True
+                break
+        if certainty == False:
+            return True
+        if message_found == True:
+            return True
+        else:
+            return False
+        
+    def isListEmpty(self):
+        isEmpty = False
+        if len(self.theList) == 0:
+            isEmpty = True
+        else:
+            isEmpty = False
+
+        return isEmpty 
+
+class ACE_S_LIST_CHECKER:
+
+    def __init__(self, theList,theDict):
+        self.theList = theList
+        self.theDict = theDict
+
+
+    def checker (self, group_index, message_index, certainty):
+        message_found = False
+        crresp_search = test_yaml['test']['groups'][i]['messages'][j]['CRRESP']
+        k = 0
+        for l in range(0,len(self.theList)):
+
+                if self.theList[l].address == search_address:
+
+                    log_file.write(self.theList[l].srcline)
+                    k = l
+
+                    if len(self.theList) != 0:
+                        if self.theList[l+1].crresp == crresp_search :
+                            log_file.write("\t\t" + self.theList[l+1].srcline)
+                            message_found = True
+                            break
+                        else:
+                            print("Test sequence " + str(crresp_search))
+                            print("Logs " + str(self.theList[l+1].crresp))
+                            print(certainty)
+                         
+                            break
+        if certainty == False:
+            return True
+
+        if message_found == True:
+            del self.theList[k]
+            del self.theList[k]
+            return True
+        else:
+            return False
+
+    def isListEmpty(self):
+        isEmpty = False
+        if len(self.theList) == 0:
+            isEmpty = True
+        else:
+            isEmpty = False
+
+        return isEmpty          
+       
+def getIndexLists():
+   indexes = []
+   buffer_access_dict[test_yaml['test']['test_buffer']] = []
+   groups = test_yaml['test']['groups']
```

Comment:
`groups` could be used in the following lines instead of `test_yaml['test']['groups']`

Reply:
okay

Reply:
done

Reply:
I still see `test_yaml['test']['groups']` - or did I misunderstand something?

Reply:
(anyway, it's no big deal, we can leave it as it is)